src/lib/core/evio.c

```diff
@@ -404,19 +404,35 @@ evio_service_bind(struct evio_service *service, const char *uri)
 /** It's safe to stop a service which is not started yet. */
 void
 evio_service_stop(struct evio_service *service)
+{
+	evio_service_deactivate(service);
+	if (service->ev.fd >= 0) {
+		int fd = service->ev.fd;
+		if (close(service->ev.fd) < 0) {
+			diag_set(SocketError, sio_socketname(fd), "close");
+			diag_log();
+			diag_clear(diag_get());
```

Comment:
There is `say_error()` function for that. The same below.

Reply:
Fixed